content/en/blog/2019/monitoring-external-service-traffic/index.md

```diff
@@ -0,0 +1,360 @@
+and labels emitted are conditional on whether the virtual outbound or explicit port/protocol
+listener is being hit.
+
+### Using the augmented metrics
+
+To capture all external service traffic in either of the cases (`BlackHole` or
+`Passthrough`), you will need to monitor `istio_requests_total` and
+`istio_tcp_connections_closed_total` metrics. Depending upon the Envoy listener
+type i.e. TCP proxy or HTTP proxy that gets invoked, one of these metrics
+will be incremented.
+
+Additionally, in case of a TCP proxy listener in order to see the IP address of
+the external service that is blocked or allowed via `BlackHole` or `Passthrough`
+cluster, you will need to add the `destination_ip` label to the
+`istio_tcp_connections_closed_total` metric. In this scenario, the host name of
+the external service is not captured. This label is not added by default and can
+be easily added by augmenting the Istio configuration for attribute generation
+and Prometheus handler.
+
+#### `PassthroughCluster` metrics
+
+This section explains the metrics and the labels emitted based on the listener
+type invoked in Envoy.
+
+* HTTP proxy listener: This happens when the port of the external service is
+  same as one of the service ports defined in the cluster. In this scenario,
+  when the `PassthroughCluster` is hit, `istio_requests_total` will get increased
+  like this:
+
+  {{< text json >}}
+    {
+      "metric": {
+        "__name__": "istio_requests_total",
+        "connection_security_policy": "unknown",
+        "destination_app": "unknown",
+        "destination_principal": "unknown",
+        "destination_service": "httpbin.org",
+        "destination_service_name": "PassthroughCluster",
+        "destination_service_namespace": "unknown",
+        "destination_version": "unknown",
+        "destination_workload": "unknown",
+        "destination_workload_namespace": "unknown",
+        "instance": "100.96.2.183:42422",
+        "job": "istio-mesh",
+        "permissive_response_code": "none",
+        "permissive_response_policyid": "none",
+        "reporter": "source",
+        "request_protocol": "http",
+        "response_code": "200",
+        "response_flags": "-",
+        "source_app": "sleep",
+        "source_principal": "unknown",
+        "source_version": "unknown",
+        "source_workload": "sleep",
+        "source_workload_namespace": "default"
+      },
+      "value": [
+        1567033080.282,
+        "1"
+      ]
+    }
+  {{< /text >}}
+
+  Note that the `destination_service_name` label is set to `PassthroughCluster` to
+  indicate that this cluster was hit and the `destination_service` is set to the
+  host of the external service.
+
+* TCP proxy virtual listener - If the external service port doesn't map to any
+  HTTP based service ports within the cluster, this listener is invoked and
+  `istio_tcp_connections_closed_total` is the metric that will be increased:
```

Comment:
presumably, `istio_tcp_connections_opened_total` will also be increased (and the bytes metrics too)?

Reply:
`istio_tcp_connections_opened_total` and other metrics don't get increased in all cases specially for `BlackHole` (there's no cluster) as a connection is never created hence no data is even received/sent.

Reply:
Added details around when these metrics will and will not get updated.